### ISSUE TITLE
使用官方微调脚本时遇到了KeyError('loss')问题

### ISSUE THREAD

**Issue by ShaohonChen** [THUDM/GLM-4#739]
### System Info / 系統信息

感谢zhipu的开源工作！

我在使用官方脚本进行微调时，遇到了KeyError('loss')的报错，这个似乎发生Eval阶段调用“prediction_step”函数时。我尝试了切换transformers版本以及切换更早的模型，但是并没有解决这个问题。我放一下我的微调报错信息

可以在这里看到我的错误日志：https://swanlab.cn/@ShaohonChen/finetune/runs/yndc6hkqjocy4xtri7ulw/log

为了方便检索我也粘贴出来

```shell
File "/home/chenshaohon/projects/GLM-4/finetune/finetune.py", line 480, in <module>
app()
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/typer/main.py", line 339, in __call__
raise e
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/typer/main.py", line 322, in __call__
return get_command(self)(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/click/core.py", line 1161, in __call__
return self.main(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/typer/core.py", line 677, in main
return _main(
^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/typer/core.py", line 195, in _main
rv = self.invoke(ctx)
^^^^^^^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/click/core.py", line 1443, in invoke
return ctx.invoke(self.callback, **ctx.params)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/click/core.py", line 788, in invoke
return __callback(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/typer/main.py", line 697, in wrapper
return callback(**use_params)
^^^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/projects/GLM-4/finetune/finetune.py", line 439, in main
trainer.train()
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/transformers/trainer.py", line 2245, in train
return inner_training_loop(
^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/transformers/trainer.py", line 2627, in _inner_training_loop
self._maybe_log_save_evaluate(
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/transformers/trainer.py", line 3096, in _maybe_log_save_evaluate
metrics = self._evaluate(trial, ignore_keys_for_eval)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/transformers/trainer.py", line 3045, in _evaluate
metrics = self.evaluate(ignore_keys=ignore_keys_for_eval)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/transformers/trainer_seq2seq.py", line 197, in evaluate
return super().evaluate(eval_dataset, ignore_keys=ignore_keys, metric_key_prefix=metric_key_prefix)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/transformers/trainer.py", line 4154, in evaluate
output = eval_loop(
^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/transformers/trainer.py", line 4348, in evaluation_loop
losses, logits, labels = self.prediction_step(model, inputs, prediction_loss_only, ignore_keys=ignore_keys)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/projects/GLM-4/finetune/finetune.py", line 72, in prediction_step
loss, generated_tokens, labels = super().prediction_step(
^^^^^^^^^^^^^^^^^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/transformers/trainer_seq2seq.py", line 356, in prediction_step
loss = (outputs["loss"] if isinstance(outputs, dict) else outputs[0]).detach().mean()
~~~~~~~^^^^^^^^
File "/home/chenshaohon/miniconda3/envs/glm4-finetune/lib/python3.11/site-packages/transformers/utils/generic.py", line 445, in __getitem__
return inner_dict[k]
~~~~~~~~~~^^^
KeyError('loss')
```

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

进入到finetune文件夹下使用如下命令（为了便于复现我将命令改成了单卡）

CUDA_VISIBLE_DEVICES=0 OMP_NUM_THREADS=1 torchrun --standalone --nnodes=1 --nproc_per_node=1  finetune.py  data/AdvertiseGen/  THUDM/GLM-4-9B-0414/  configs/lora.yaml

### Expected behavior / 期待表现

能正常运行而非报错

**Comment by ShaohonChen** [comment]
文档和配置文件中存在两处小错误，我也提交了修改的PR

#740 
#738

**Comment by zRzRzRzRzRzRzR** [comment]
我推送了一个更改，请查看

**Comment by ShaohonChen** [comment]
可以了，我这边测试没问题，是否需要我在finetune文件夹下放一个aplaca中文的缩小版本的案例训练集？方便大家实际上手训练时能快速使用。我看之前的issue中有提到找不到数据集的问题  @zRzRzRzRzRzRzR

**Comment by zRzRzRzRzRzRzR** [comment]
我们不会在仓库中推送完整的数据集，因为这会让github仓库增加很多不必要的最小文件。在readme中，我们给出了详细的构建方法，开发者可以根据readme中的格式自行构建

**Comment by 159357hou** [comment]
请问最后是如何解决的呢，我也遇到了同样的问题？
